### PR TITLE
Home Page Object addition 10.6.20

### PR DESCRIPTION
**Pull request by agrant23**

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -1,5 +1,10 @@
 
 import os
 
-chromedriver_path = os.path.dirname(os.path.realpath(__file__)) + '/resources/chromedriver.exe'
+windows_chromedriver = 'chromedriver.exe'
```
Comment (MNadolsky) [inline config.py]: Revert this change. My understanding is that if you change the chromedriver filename it will cause errors, and since mac and linux chromedriver files are identically named we're unlikely to implement multi-OS support this way.

File: config.py [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
 
+chromedriver_path = os.path.dirname(os.path.realpath(__file__)) + '/resources/' + windows_chromedriver
+
+
+homepage_url = 'https://www.coinbase.com/'
```
Comment (MNadolsky) [inline config.py]: name these `coinbase_homepage_url` and `coinbase_signin_page_url` because when we add more websites we will need to differentiate.

File: config.py [hunk 3]
```diff
@@ -1,5 +1,10 @@
...
+signInPage_url = 'https://www.coinbase.com/signin'
```
Comment (MNadolsky) [inline config.py]: This will be changed by the comment above, but, an additional comment, it's a bad idea to mix camel case and snake case. Pick one.

File: Home_Page.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    # via cellphone; it is optional to forego this authentication, so 
+    # cookies are added below to do this. It is currently unclear which cookie controls 
+    # the authentication. So, I just signed in, checked the option to not ask for
+    # the text code again, then signed out and grabbed all the cookies. Below I loaded 
+    # all these cookies which include the cookie/s necessary to forego this second 
+    # authentication.
+
+
+    def click_home_page_sign_in_button(self):
+        home_page_sign_in_button_loc = (By.XPATH,"//a[@title='Sign in']")
```
Comment (MNadolsky) [inline Home_Page.py]: Pull the locator out of the method so it can be used by other methods.

File: Home_Page.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+from selenium import webdriver
```
Comment (MNadolsky) [inline Home_Page.py]: Put this file in a directory named `coinbase/`. Rename the file to be `homepage.py`. Going to try to stick to PEP8 basics. We should stick to this for all filenames. Consider changing any you see that don't comply when you're working on a related ticket.

https://www.python.org/dev/peps/pep-0008/#package-and-module-names

The name of the file should be the url extension when it makes sense (in the case of coinbase, "signin," "dashboard," etc.).

imports should read `import pages.coinbase.homepage`, `pages.coinbase.signin`. Other sites should read, e.g., `pages.wikipedia.homepage`, `pages.wikipedia.wiki`. etc. I'll add this to the upcoming confluence page.

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,5 @@
 # ThePlatform
+Secure.py
```
Comment (MNadolsky) [inline .gitignore]: What is this? There is no secure.py. We removed it in favor of secure/

Unless something is going on I'm not thinking about, remove this.

**Review by MNadolsky — CHANGES_REQUESTED** [review]
Please make the requested changes and update the PR.

File: pages/coinbase/homepage.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    # cookies are added below to do this. It is currently unclear which cookie controls 
+    # the authentication. So, I just signed in, checked the option to not ask for
+    # the text code again, then signed out and grabbed all the cookies. Below I loaded 
+    # all these cookies which include the cookie/s necessary to forego this second 
+    # authentication.
+
+    def home_page_sign_in_button_loc(self): return (By.XPATH,"//a[@title='Sign in']")
+    def click_home_page_sign_in_button(self):
+        self.driver.find_element(*self.home_page_sign_in_button_loc()).click()
+        for cookie in secure.cookies.coinbase: self.driver.add_cookie(cookie)
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: Remove this line and the accompanying comment above. The cookie injection should be a method in the /signin object, and it should either be an optional parameter of clicking the sign in button on that page, an optional parameter of a sign in workflow, or both. (When you move this, make sure the accompanying comment goes above the this line, not above the function it's in.)

You don't want it in this method, and you definitely don't want it to be mandatory, because it blocks us from testing two-factor auth.

File: pages/coinbase/homepage.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: remove any imports you're not using anymore. (all imports you're not using, if it's more than just this one)

File: pages/coinbase/homepage.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+    # coinbase has two-factor authentication which involves sending a text
+    # via cellphone; it is optional to forego this authentication, so 
+    # cookies are added below to do this. It is currently unclear which cookie controls 
+    # the authentication. So, I just signed in, checked the option to not ask for
+    # the text code again, then signed out and grabbed all the cookies. Below I loaded 
+    # all these cookies which include the cookie/s necessary to forego this second 
+    # authentication.
+
+    def home_page_sign_in_button_loc(self): return (By.XPATH,"//a[@title='Sign in']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: This doesn't need to be a function. just `sign_in_button_loc = (By.XPATH, "//a[@title='Sign in']")`

Also, don't use redundant naming. This function is `Home_Page.home_page_sign_in_button_loc()`; don't name functions in such a way that you have to type something (like "home page") twice.

**Review by MNadolsky — CHANGES_REQUESTED** [review]

File: pages/coinbase/homepage.py (hunk 9 above)
Comment (agrant23) [reply pages/coinbase/homepage.py]: There are two sign in buttons one on the home page and one on the sign in page. But I guess since they are in separate files I don't need to distinguish them.

File: pages/coinbase/homepage.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+import secure.cookies
+import config
+
+
+#sign_in_button_loc = (By.XPATH,"//a[@title='Sign in']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: remove this comment.

**Review by MNadolsky — CHANGES_REQUESTED** [review]

File: pages/coinbase/homepage.py (hunk 9 above)
Comment (MNadolsky) [reply pages/coinbase/homepage.py]: The class name distinguishes them already. to use the button on the home page, you have to use HomePage.sign_in_button. to use the button on the sign_in_page you have to use SigninPage.sign_in_button.

**Review by MNadolsky — APPROVED** [review]
approved.
